dask/dataframe/io/parquet/arrow.py

```diff
@@ -652,7 +652,7 @@ def read_partition(
         # to categorigal manually for integer types.
         if partitions and isinstance(partitions, list):
             for partition in partitions:
-                if df[partition.name].dtype.name != "category":
+                if len(partition.keys) and df[partition.name].dtype.name != "category":
```

Comment:
The original logic here was assuming that we want to convert all hive/directory-partitioned columns to `"category"` if pyarrow hasn't already done this for us. However, we can only do this if we know all possible categories (stored in `partition.keys`).

If the user specifies a partitioning schema with a non-categorical type, `partition.keys` will be empty, and we should **not** try to convert to a `"category"` dtype.